File: app/rest/api_1_0/viewmodels/AuthenticationViewmodels.scala
package rest.api_1_0.viewmodels

/**
  * View Models for Authentication.
  * Created by Christian on 04.05.2016.
  */
object AuthenticationViewModels {

  case class LoginRequest(username: String,
                          password: String) extends ViewModel

  case class LoginResult(success: Boolean,
                         authenticationStatus: AuthenticationStatus) extends ViewModel

  case class LogoutResult(success: Boolean,
                          authenticationStatus: AuthenticationStatus) extends ViewModel

  case class AuthenticationStatusUser(id: String,
                                      username: String,
                                      displayName: String,
                                      fullName: Option[String],
                                      avatar: Option[String],
                                      groups: Option[Seq[String]]) extends ViewModel

  case class AuthenticationStatus(authenticated: Boolean,
                                  sessionId: Option[String],
                                  user: Option[AuthenticationStatusUser],
                                  globalPermissions: Seq[String]) extends ViewModel

  case class GetAuthenticationStatusResult(success: Boolean,
                                           authenticationStatus: AuthenticationStatus) extends ViewModel

}

File: frontend/viewmodels/AuthenticationViewModels.ts
import {ApiResult} from "./GeneralViewModels"

/**
 * ViewModels for Authentication
 */

export class LoginRequest {
    constructor(public username:string,
                public password:string) {
    }
}

export interface AuthenticationStatusUser {
    id:string
    username:string
    email:string
    displayName:string
    fullName?:string
    avatar?:string,
    groups?:Array<string>
}

export interface AuthenticationStatus {
    authenticated:boolean
    sessionId?:string
    user?:AuthenticationStatusUser
    globalPermissions:Array<string>
}

export interface AuthenticationResult extends ApiResult {
    authenticationStatus:AuthenticationStatus
}

export interface LoginResult extends AuthenticationResult {
}

export interface LogoutResult extends AuthenticationResult {
}

export interface GetAuthenticationStatusResult extends AuthenticationResult {
}

export class RegisterRequest {
    constructor(public username:string,
                public email:string,
                public password:string) {
    }
}

export interface RegisterResult extends AuthenticationResult {
}